### PR TITLE
[SPARK-36258][PYTHON] Exposing functionExists in pyspark sql catalog

### PR DESCRIPTION
**Pull request by dominikgehl**
### What changes were proposed in this pull request?
Exposing functionExists in pyspark sql catalog

### Why are the changes needed?
method was available in scala but not pyspark

### Does this PR introduce _any_ user-facing change?
Additional method

### How was this patch tested?
Unit tests

### CODE REVIEW

File: python/pyspark/sql/tests/test_catalog.py [hunk 1]
```diff
@@ -135,7 +135,7 @@ def test_list_functions(self):
             self.assertEqual(functions, functionsDefault)
 
             with self.function("func1", "some_db.func2"):
-                spark.catalog.registerFunction("temp_func", lambda x: str(x))
+                spark.udf.register("temp_func", lambda x: str(x))
```
Comment (dominikgehl) [inline python/pyspark/sql/tests/test_catalog.py:138]: to get rid of the deprecated warning in the unit tests

Comment (HyukjinKwon) [reply python/pyspark/sql/tests/test_catalog.py:138]: nice

**Review by HyukjinKwon — APPROVED** [review]

File: python/pyspark/sql/catalog.py [hunk 2]
```diff
@@ -132,6 +132,30 @@ def listFunctions(self, dbName=None):
...
+            name of the function to check existence
+        dbName : str, optional
+            name of the database to check function existence in.
+            If no database is specified, the current database is used
+
+        Returns
+        -------
+        bool
+            Indicating whether the function exists
+
```
Comment (HyukjinKwon) [inline python/pyspark/sql/catalog.py:153]: @dominikgehl since we're here, could we add some simple examples here as well?

Comment (dominikgehl) [reply python/pyspark/sql/catalog.py:153]: Sure. Done

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**